src/common/LRUCache.cpp

```diff
@@ -40,6 +41,11 @@ class LRUCache {
             _max_size(max_size) {
     }
 
+    void adjustSize(size_t newSize) {
```

Comment:
Need to check if the number of cache item is greater than the target size, if so need to pop them up